app/src/main/kotlin/com/github/gotify/Settings.kt

```diff
@@ -26,12 +26,21 @@ internal class Settings(context: Context) {
     var serverVersion: String
```

Comment:
> Because of the new cert retrieval logic, upgrades from previous app versions will lead to ignored ca certs

Would it be difficult to make this backwards compatible?

Reply:
I think 1 is preferred but 2 is also fine.

Reply:
Understood, I will try that

Reply:
It's now implemented in https://github.com/gotify/android/pull/344/commits/a3dd80cd20b47e705231398bd23d185fe91ce793.  
The check is performed before client creation, but because there are several clients created at the same time, the first request fails direct after cert migration.  
Clicking retry then fixed it as the client then found the certificate.

A solution could be interactive dialog to restart the app or something, but then the migration does not work in the background without opening the app. With the current implementation the user can just update the app and when the websocket starts, it can (hopefully) migrate it by itself.